core/metadata.js

```diff
@@ -188,6 +189,18 @@ function invariants (doc /*: Metadata */) {
   }
 }
 
+function copyIfNull (doc /*: Metadata */, fields /*: Array<string> */, was /*: Metadata */) {
```

Comment:
I even wonder whether we should actually list fields by hand instead of generalizing the logic inside this function.
E.g.:

- If no was, return
- Copy missing tags/remote/ino fields from was to doc (with default [] for tags)
- If doc is a file, copy missing executable field from was (honoring 'na')
  * if sameBinary, copy missing size, class & mime from was

And just call it at the beginning of every merge method...

Reply:
We could maybe even include other fields this way (`_rev`, `moveFrom`...).